src/utils.ts

```diff
@@ -0,0 +1,49 @@
+import { Readable } from "stream";
+import { Block } from "./types";
+
+export const sleep = (pause: number) =>
+  new Promise((resolve) => setTimeout(resolve, pause));
+
+export function prependZeroes(number: number) {
+  return number.toString().padStart(12, "0");
+}
+
+export async function parseBody<T>(stream: Readable): Promise<T> {
+  const contents = await streamToString(stream);
+  const parsed: T = JSON.parse(contents, renameUnderscoreFieldsToSnakeCase);
+  return parsed;
+}
+
+// the function got from
+// https://github.com/aws/aws-sdk-js-v3/issues/1877#issuecomment-755387549
+function streamToString(stream: Readable): Promise<string> {
+  return new Promise((resolve, reject) => {
+    const chunks = [];
+    stream.on("data", (chunk) => chunks.push(chunk));
+    stream.on("error", reject);
+    stream.on("end", () => resolve(Buffer.concat(chunks).toString("utf8")));
+  });
+}
+
+// function renames all fields in the nested object
+// from underscore_style to snakeCase
+function renameUnderscoreFieldsToSnakeCase(name, value) {
```

Comment:
* [ ] snake_case_is_this and camelCaseIsThis, please, adjust the function name and the comment, as you `renameSnakeCaseFieldsToCamelCase`
* [ ] `name` argument is not used, so it should be removed

Reply:
As for `name`. This function is reviver for `JSON.parse` and according to the docs, it will get the `key` as the first argument. Removing is not an option for us in this case

ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse?retiredLocale=uk#using_the_reviver_parameter

Reply:
I was sure that this is CamelCase (with first capital letter), my bad.

Reply:
That is CapitalCase, there is also kebab-case, fyi 😄

Reply:
> As for name. This function is reviver for JSON.parse and according to the docs, it will get the key as the first argument. Removing is not an option for us in this case

Either document that and rename it to `_` or just use a lambda function when you pass it to `JSON.parse` to discard the first argument. (I would go with the second option, to be honest)